lib/Serialization/Serialization.cpp

```diff
@@ -3088,7 +3088,6 @@ class Serializer::DeclSerializer : public DeclVisitor<DeclSerializer> {
                                rawAccessLevel, proto->getInherited().size(),
                                inheritedAndDependencyTypes);
 
-    const_cast<ProtocolDecl*>(proto)->createGenericParamsIfMissing();
     writeGenericParams(proto->getGenericParams());
```

Comment:
Optional cleanup: we gain nothing by serializing or deserializing the generic parameter list of a protocol, since we can always re-derive it from scratch.